packages/mui-base/src/AutocompleteUnstyled/useAutocomplete.js

```diff
@@ -326,10 +333,10 @@ export default function useAutocomplete(props) {
       return;
     }
 
-    const prev = listboxRef.current.querySelector('[role="option"].Mui-focused');
+    const prev = listboxRef.current.querySelector(`[role="option"].${_classNamePrefix}-focused`);
     if (prev) {
-      prev.classList.remove('Mui-focused');
-      prev.classList.remove('Mui-focusVisible');
+      prev.classList.remove(`${_classNamePrefix}-focused`);
+      prev.classList.remove(`${_classNamePrefix}-focusVisible`);
```

Comment:
Currently, Joy UI is using `Joy` as a prefix for generating className.

We could remove this internal config in the future once we have the decision on https://github.com/mui/material-ui/discussions/34069